util/gogc.go

```diff
@@ -40,10 +40,64 @@ func SetGOGC(val int) {
 	}
 	debug.SetGCPercent(val)
 	metrics.GOGC.Set(float64(val))
-	atomic.StoreInt64(&gogcValue, int64(val))
+	gogcValue.Store(int64(val))
 }
 
 // GetGOGC returns the current value of GOGC.
 func GetGOGC() int {
-	return int(atomic.LoadInt64(&gogcValue))
+	return int(gogcValue.Load())
+}
+
+// GOGCStatSampler is a sampler for GOGC.
+type GOGCStatSampler struct {
+	last struct {
+		now          int64
+		lastGC       time.Time
+		gcPauseTotal uint64
+	}
+	GCRunCount uint64
+	exitChan   chan struct{}
+}
+
+// NewRuntimeStatSampler returns a sampler for GOGC.
+func NewRuntimeStatSampler() *GOGCStatSampler {
+	return &GOGCStatSampler{
+		exitChan: make(chan struct{}),
+	}
+}
+
+var gc debug.GCStats
+
+func (gss *GOGCStatSampler) sample() {
+	debug.ReadGCStats(&gc)
+	now := time.Now().UnixNano()
+	dur := float64(now - gss.last.now)
+	gcPauseRatio := float64(uint64(gc.PauseTotal)-gss.last.gcPauseTotal) / dur
+	if gss.last.lastGC.Before(gc.LastGC) {
+		gcIntervals := gc.LastGC.Sub(gss.last.lastGC)
+		gss.last.lastGC = gc.LastGC
+		metrics.GCIntervals.Set(float64(gcIntervals.Nanoseconds()))
```

Comment:
If you want to know when GC happens, it's already there in the grafana...
And from the image the `GCIntervals` are obvious

Reply:
You are right. I will remove it.

Reply:
GC durations? I think it is different with GC intervals.